src/KafkaSchemaRegistryApiClient.php

```diff
@@ -126,7 +127,7 @@ public function checkSchemaCompatibilityForVersion(
                        sprintf('compatibility/subjects/%s/versions/%s', $subjectName, $version),
                        $this->createRequestBodyFromSchema($schema)
                    );
-        } catch (SubjectNotFoundException $e) {
+        } catch (SubjectNotFoundException|VersionNotFoundException $e) {
```

Comment:
I think that the compatibility check shouldn't be successful (return true), if I pass a non-existing version on an existing subject/schema, as opposed to a non-existing subject/schema. E.g. let's say there is an existing version 1 + 2 of schema x in the registry, now I check the compatibility for a new version of the schema x which is not compatible with version 1 + 2, but instead of passing 1 or 2 to this method, I pass 3. Why do you think we should return true in such a case?

Reply:
The newest Karapace version now returns `VersionNotFoundException` if the subject does not exist, i left the old exception for compatibility reasons with older versions

Reply:
👍 I see, then I suggest we wait for the outcome. But if this is intended behaviour I would rather have it not handled in this method and do a compatibility check by first checking if the schema exists and then if the new version is compatible with an existing one instead of falsely indicating that the new version is compatible in the case I mentioned above.

Reply:
You were faster :smile: i think with the newest commit, the check should be clean, if the `latest` version throws `VersionNotFoundException` it is actually a `SubjectNotFoundException` :v: